packages/curve-ui-kit/src/shared/ui/LargeTokenInput.tsx

```diff
@@ -209,7 +226,14 @@ export const LargeTokenInput = ({ tokenSelector, maxBalance, message, isError =
    */
   useEffect(() => {
```

Comment:
@0xAlunara to answer your question I think the proper fix would be to lift the state out of the component and change the balance as soon as we get the balance instead of using an effect.

A fix without too much refactoring would be to add `useRef` with the balance or the balance input